src/components/AdditionalSpecies.jsx

```diff
@@ -0,0 +1,196 @@
+import React, { useState } from 'react';
+import { BsPersonFill, BsArrowDown } from 'react-icons/bs';
+import { nanoid } from 'nanoid';
+import {
+  Heading,
+  VStack,
+  SimpleGrid,
+  GridItem,
+  Button,
+  Table,
+  Thead,
+  Tbody,
+  Tr,
+  Th,
+  Td,
+  Input,
+  NumberInput,
+  NumberInputField,
+  Checkbox,
+  FormLabel,
+  FormControl,
+  IconButton,
+  Container,
+  Icon,
+} from '@chakra-ui/react';
+import { DeleteIcon } from '@chakra-ui/icons';
+import Select from 'react-select';
+import PropTypes from 'prop-types';
+
+const rows = [
+  { name: 'Beach Cast', total: 0, isChecked: false, isVisible: 'hidden', isDisabled: true },
+];
+
+const AdditionalSpecies = ({ options }) => {
+  const [species, setSpecies] = useState(rows);
+  const [allChecked, setAllChecked] = useState(false);
+  const [disabled, setDisabled] = useState(true);
+  const [option, setOption] = useState('');
+
+  const findChecked = value => {
+    return value.isChecked;
+  };
+
+  const handleAddTotalChange = (value, id) => {
+    const newTotalData = [...species];
+    newTotalData[id].total = value;
+
+    setSpecies(newTotalData);
+  };
+
+  const handleAllChecked = () => {
+    const newCheckedData = [...species];
+    if (allChecked) {
+      for (let i = 1; i < newCheckedData.length; i += 1) {
+        newCheckedData[i].isChecked = false;
+      }
+    } else {
+      for (let i = 1; i < newCheckedData.length; i += 1) {
+        newCheckedData[i].isChecked = true;
+      }
+    }
+    setDisabled(!newCheckedData.some(findChecked));
+    setAllChecked(!allChecked);
+    setSpecies(newCheckedData);
+  };
+
+  const handleRowCheckedItems = id => {
+    const newCheckedData = [...species];
+    newCheckedData[id].isChecked = !newCheckedData[id].isChecked;
+    setDisabled(!newCheckedData.some(findChecked));
+    setSpecies(newCheckedData);
+  };
+
+  const handleAddRow = () => {
+    if (option !== '') {
+      const newData = {
+        name: option.label,
+        total: 0,
+        isChecked: false,
+        isVisible: 'visible',
+        isDisabled: false,
+      };
+      const newSpecies = [...species, newData];
+
+      setOption('');
+      setSpecies(newSpecies);
+    }
+  };
+
+  const handleDeleteRows = () => {
+    const newSpecies = species.filter(row => !row.isChecked);
+    setDisabled(!newSpecies.some(findChecked));
+    setSpecies(newSpecies);
+  };
+
+  const createTable = m => {
+    return m.map((row, index) => (
+      <Tr height="72px" key={nanoid()}>
```

Comment:
React expects the `key` value to be stable, as it uses it to determine if a component should be rerendered. The [Nano ID documentation recommends against using](https://github.com/ai/nanoid#react) `nanoid` in this way, as the value is not stable and can cause unnecessary rerenders which hurt performance.

For now, I would remove `nanoid` altogether and use `index` from the map. Once actual data is used to render the components, the key should be switched to the use unique ID of that data.

You can read more about how React uses keys for rendering [here](https://reactjs.org/docs/lists-and-keys.html).

Reply:
I created a counter to increase a number variable everytime a row was added.